src/jekyll/spark-submit.md

```diff
@@ -0,0 +1,599 @@
+-rw-rw-r--  1 datalayer datalayer 11238 Jan 13 17:57 log4j.properties
+-rw-rw-r--  1 datalayer datalayer  3180 Jan 13 17:57 mapred-site.xml
+-rw-rw-r--  1 datalayer datalayer  1280 Jan 13 17:57 __spark_conf__.properties
+-rw-rw-r--  1 datalayer datalayer  2316 Jan 13 17:57 ssl-client.xml
+-rw-rw-r--  1 datalayer datalayer  2268 Jan 13 17:57 ssl-server.xml
+-rw-rw-r--  1 datalayer datalayer  4131 Jan 13 17:57 yarn-site.xml
+```
+
+```console
+ __spark_libs__320488134561245276/
+-rw-rw-r--  1 datalayer datalayer    69409 Jan 13 17:57 activation-1.1.1.jar
+-rw-rw-r--  1 datalayer datalayer   445288 Jan 13 17:57 antlr-2.7.7.jar
+-rw-rw-r--  1 datalayer datalayer   302248 Jan 13 17:57 antlr4-runtime-4.5.3.jar
+...
+-rw-rw-r--  1 datalayer datalayer  8270638 Jan 13 17:57 spark-catalyst_2.11-2.2.0-k8s-0.5.0.jar
+...
+-rw-rw-r--  1 datalayer datalayer    46566 Jan 13 17:57 spark-unsafe_2.11-2.2.0-k8s-0.5.0.jar
+-rw-rw-r--  1 datalayer datalayer   696193 Jan 13 17:57 spark-yarn_2.11-2.2.0-k8s-0.5.0.jar
+...
+-rw-rw-r--  1 datalayer datalayer    35518 Jan 13 17:57 zjsonpatch-0.3.0.jar
+-rw-rw-r--  1 datalayer datalayer   792964 Jan 13 17:57 zookeeper-3.4.6.jar
+```
+
+In case of spark-submit (client or cluster mode), the process is the same and generates the same folder structure.
+
+```console
+-rw-r--r--   1 datalayer supergroup      50608 2018-01-13 18:13 /user/datalayer/.sparkStaging/application_1515858696776_0016/__spark_conf__.zip
+-rw-r--r--   1 datalayer supergroup  232012522 2018-01-13 18:13 /user/datalayer/.sparkStaging/application_1515858696776_0016/__spark_libs__3393918138649389825.zip
+-rw-r--r--   1 datalayer supergroup     619037 2018-01-13 18:13 /user/datalayer/.sparkStaging/application_1515858696776_0016/hbase-common-1.4.0.jar
+-rw-r--r--   1 datalayer supergroup    1991211 2018-01-13 18:13 /user/datalayer/.sparkStaging/application_1515858696776_0016/spark-examples_2.11-2.2.0-k8s-0.5.0.jar
+```
+
+*[TBC] Explaine this log: 2018-01-13 12:15:27 WARN  SparkConf:66 - In Spark 1.0 and later spark.local.dir will be overridden by the value set by the cluster manager (via SPARK_LOCAL_DIRS in mesos/standalone and LOCAL_DIRS in YARN).*
+
+# Notes
+
+## K8S compared to YARN
+
+YARN resource manager uploads from the submssion process to HDFS the jars and files. The YARN framework has built-in capability to use in the YARN containers the defined resources.
+
+K8S reource manager delegates to the init container the download of the jars and files. It reads the given spark.jars property and get them directly from HDFS for example.
+
++ Deps distribution: Download deps only if needed.
++ spark-core vs init-container: ...
++ Need HDFS: ...
++ Spark susbmission not connected on HDFS: ...
++ Other distribution cache: ...
++ Hadoop conf dir: Hadoop conf must not be present before hand on nodes and can be mounted from configmaps.
+
+### Submit in client mode from a client with restricted network access
+
+This is not possible. When the executor is invoked.
```

Comment:
this is a little vague -- can you explain what restrictions don't work?  obviously, if there is no communication out from the client, nothing is going to work.  But one might expect that you only need some very limited communication from the client.  Eg., only client to driver pod, but not client to executors.